spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/OpenTelemetryConfigurations.java

```diff
@@ -168,6 +182,105 @@ OtelHttpServerHandler otelHttpServerHandler(OpenTelemetry openTelemetry) {
 					new DefaultHttpServerAttributesExtractor());
 		}
 
+		@Configuration(proxyBeanMethods = false)
+		static class PropagationConfiguration {
+
+			@Bean
+			@ConditionalOnMissingBean
+			@ConditionalOnProperty(value = "management.tracing.propagation", havingValue = "B3", matchIfMissing = true)
+			TextMapPropagator b3TextMapPropagator() {
+				return B3Propagator.injectingSingleHeader();
+			}
+
+			@Configuration(proxyBeanMethods = false)
+			@ConditionalOnProperty(value = "management.tracing.baggage.enabled", havingValue = "false")
+			static class NoBaggagePropagatorConfiguration {
+				@Bean
+				@ConditionalOnMissingBean
+				@ConditionalOnProperty(value = "management.tracing.propagation", havingValue = "W3C")
+				TextMapPropagator w3cTextMapPropagator() {
+					return W3CTraceContextPropagator.getInstance();
+				}
+
+			}
+
+			@Bean
+			@ConditionalOnProperty(value = "management.tracing.propagation", havingValue = "AWS")
+			@ConditionalOnClass(AWSPropagation.class)
+			TextMapPropagator awsTextMapPropagator() {
```

Comment:
To provide as much type information as possible to the bean factory, the signature should show that it returns an `AwsXrayPropagator`.